### PR TITLE
Added redshift.get_cluster_credentials

### PR DESCRIPTION
**Pull request by dreamorosi**
Implemented as described in #3609.

### CODE REVIEW

**Comment by dreamorosi** [comment]
@bblommers Latest CI build failed for Python 2.7 on already existing tests. The ones that I have added in the PR are all passed.
Thoughts?

File: moto/redshift/models.py [hunk 1]
```diff
@@ -941,6 +943,36 @@ def delete_tags(self, resource_name, tag_keys):
         resource = self._get_resource_from_arn(resource_name)
         resource.delete_tags(tag_keys)
 
+    @staticmethod
+    def _generate_random_alphanum():
```
Comment (bblommers) [inline moto/redshift/models.py:947]: Let's move this into utilities.utils.py, there's a similar method there already - makes sense to have them in the same place.

File: moto/redshift/models.py [hunk 2]
```diff
@@ -941,6 +943,36 @@ def delete_tags(self, resource_name, tag_keys):
...
+        printable = list(string.ascii_letters + string.digits + string.punctuation)
```
Comment (bblommers) [inline moto/redshift/models.py:948]: Minor nitpick - but the method name (alphanum) is incorrect if punctuation is allowed

File: moto/redshift/models.py [hunk 3]
```diff
@@ -941,6 +943,36 @@ def delete_tags(self, resource_name, tag_keys):
...
+
+        random_alphanum = random.sample(printable, k=32)
+        random_alphanum = "".join(random_alphanum)
+        return random_alphanum
+
+    def get_cluster_credentials(
+        self, cluster_identifier, db_user, auto_create, duration_seconds
+    ):
+        try:
+            assert duration_seconds >= 900 and duration_seconds <= 3600
```
Comment (bblommers) [inline moto/redshift/models.py:959]: This can be a simple `if ..: raise `, no need for a try-except

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for the PR @dreamorosi - I've left some comments.

The failure in Py2 looks to be caused by #3613, we'll get that sorted independently.

File: moto/redshift/models.py (hunk 2 above)
Comment (dreamorosi) [reply moto/redshift/models.py]: True, my bad 😅

File: moto/redshift/models.py (hunk 1 above)
Comment (dreamorosi) [reply moto/redshift/models.py]: I removed it entirely and now using the existing one; I didn't see it at first, thanks for pointing it out.

**Review by bblommers — APPROVED** [review]
LGTM - thanks @dreamorosi!

**Pull request merged**